core/src/main/java/org/apache/iceberg/encryption/KeyEncryptionKey.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.encryption;
+
+import java.io.Serializable;
+
+public class KeyEncryptionKey implements Serializable {
+  private final String keyID;
+  private final String wrappedKey;
+  private final byte[] key;
+  private final long timestamp;
+
+  public KeyEncryptionKey(String kekID, byte[] newKek, String wrappedNewKek, long timestamp) {
+    this.keyID = kekID;
+    this.key = newKek;
+    this.wrappedKey = wrappedNewKek;
+    this.timestamp = timestamp;
+  }
+
+  public String id() {
+    return keyID;
+  }
+
+  public byte[] key() {
```

Comment:
I don't think this should hold the key. The `EncryptionManager` can provide the key multiple times by caching results from `unwrapKey`. Callers should not hold on to the key any longer than needed.

Reply:
This class is a KEK cache entry. The cache is kept in the EncryptionManager.

Reply:
In that case, why is this class public? I would expect it to be a private static class with the EncryptionManager implementaiton.